packages/core/src/Cardano/types/Certificate.ts

```diff
@@ -80,23 +30,24 @@ export interface PoolRetirementCertificate {
 
 export interface StakeDelegationCertificate {
   __typename: CertificateType.StakeDelegation;
-  address: Address;
+  rewardAccount: RewardAccount;
   poolId: PoolId;
   epoch: Epoch;
 }
 
 export interface MirCertificate {
   __typename: CertificateType.MIR;
-  address: Address;
+  rewardAccount: RewardAccount;
   quantity: Lovelace;
   pot: 'reserve' | 'treasury';
 }
 
 export interface GenesisKeyDelegationCertificate {
   __typename: CertificateType.GenesisKeyDelegation;
-  genesisHash: Hash16;
-  genesisDelegateHash: Hash16;
-  vrfKeyHash: Hash16;
+  // TODO: need to find examples of these hashes to figure out type and length
+  genesisHash: Hash32ByteBase16;
```

Comment:
https://github.com/input-output-hk/cardano-configurations/blob/26b6b6de73f90e4777602b372798bf77addcc321/network/testnet/cardano-node/config.json#L7-L15

Reply:
This verifies that GenesisHash is `Hash32ByteBase16`. Assume the same for `genesisDelegateHash` and `vrfKeyHash`?

Reply:
Yes that's a safe assumption